pkg/compact/downsample/downsample.go

```diff
@@ -167,10 +167,23 @@ func Downsample(
 						// https://github.com/thanos-io/thanos/issues/5272
 						level.Warn(logger).Log("msg", fmt.Sprintf("expected downsampled chunk (*downsample.AggrChunk) got an empty %T instead for series: %d", c.Chunk, postings.At()))
 						continue
+					} else {
+						if err := expandChunkIterator(c.Chunk.Iterator(reuseIt), &all); err != nil {
+							return id, errors.Wrapf(err, "expand chunk %d, series %d", c.Ref, postings.At())
+						}
+						aggrDataChunks := DownsampleRaw(all, ResLevel1)
+						for _, cn := range aggrDataChunks {
+							ac, ok = cn.Chunk.(*AggrChunk)
+							if !ok {
+								level.Warn(logger).Log("Not able to convert non-empty XOR chunks into 5m downsampled Aggregated chunks")
```

Comment:
I don't know if we should just throw a warning log here.
This is different from the case where the chunk has no samples, that can be safely ignored. But this one I prefer to throw error instead.

Reply:
Made changes accordingly